### PR TITLE
Move Methods from PlayerMock to HumanEntityMock

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Move Methods from HumanEntity that were defined in PlayerMock to HumanEntityMock

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 1]
```diff
@@ -1,24 +1,140 @@
...
+	private @Nullable PlayerInventoryMock inventory = null;
+	private InventoryView inventoryView;
+	private @Nullable ItemStack cursor = null;
+	private @NotNull GameMode gameMode = GameMode.SURVIVAL;
 	private @Nullable Location lastDeathLocation = new Location(new WorldMock(), 0, 0, 0);
+	protected int expLevel = 0;
+	private float saturation = 5.0F;
+	private int foodLevel = 20;
+
+
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion

```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 2]
```diff
@@ -31,5 +147,400 @@ public void setLastDeathLocation(@Nullable Location location)
...
+	public void setSaturation(float value)
+	{
+		// Saturation is constrained by the current food level
+		this.saturation = Math.min(getFoodLevel(), value);
+	}
+
+	@Override
+	public int getFoodLevel()
+	{
+		return foodLevel;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion
		return this.foodLevel;
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 3]
```diff
@@ -31,5 +147,400 @@ public void setLastDeathLocation(@Nullable Location location)
...
+	public void setExhaustion(float value)
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+
+	@Override
+	public float getSaturation()
+	{
+		return saturation;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion
		return this.saturation;
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 4]
```diff
@@ -31,5 +147,400 @@ public void setLastDeathLocation(@Nullable Location location)
...
+		return getInventory().getItemInMainHand();
+	}
+
+	@Override
+	public void setItemInHand(@Nullable ItemStack item)
+	{
+		getInventory().setItemInMainHand(item);
+	}
+
+
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion

```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 5]
```diff
@@ -31,5 +147,400 @@ public void setLastDeathLocation(@Nullable Location location)
...
+	}
+
+	@Override
+	public @Nullable InventoryView openStonecutter(@Nullable Location location, boolean force)
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+
+
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion

```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 6]
```diff
@@ -1,24 +1,140 @@
...
+		return cursor == null ? new ItemStack(Material.AIR, 0) : cursor.clone();
+	}
+
+	@Override
+	public void setItemOnCursor(@Nullable ItemStack item)
+	{
+		this.cursor = item == null ? null : item.clone();
+	}
+
+
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion

```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 7]
```diff
@@ -1,24 +1,140 @@
...
+		// reset the cursor as it is a new InventoryView
+		cursor = null;
+		inventoryView = new SimpleInventoryViewMock(this, null, inventory, InventoryType.CRAFTING);
+	}
+
+	@Override
+	public @NotNull InventoryView getOpenInventory()
+	{
+		return inventoryView;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion
		return this.inventoryView;
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 8]
```diff
@@ -1,24 +1,140 @@
...
+			Bukkit.getPluginManager().callEvent(event);
+
+			if (inventoryView.getTopInventory() instanceof InventoryMock inventoryMock)
+			{
+				inventoryMock.removeViewer(this);
+			}
+		}
+
+		// reset the cursor as it is a new InventoryView
+		cursor = null;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion
		this.cursor = null;
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 9]
```diff
@@ -1,24 +1,140 @@
...
+	}
+
+
+	@Override
+	public void closeInventory()
+	{
+		if (inventoryView instanceof PlayerInventoryViewMock)
+		{
+			InventoryCloseEvent event = new InventoryCloseEvent(inventoryView);
+			Bukkit.getPluginManager().callEvent(event);
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion
			new InventoryCloseEvent(inventoryView).callEvent();
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 10]
```diff
@@ -1,24 +1,140 @@
...
+	{
+		assertEquals(expectedGamemode, gameMode);
+	}
+
+	@Override
+	public @NotNull PlayerInventory getInventory()
+	{
+		if (inventory == null)
+		{
+			inventory = (PlayerInventoryMock) Bukkit.createInventory(this, InventoryType.PLAYER);
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java:70]: ?

**Review by Insprill — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java (hunk 10 above)
Comment (thelooter) [reply src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java:70]: Whats the Issue here. Thats how it was implemented in PlayerMock

File: src/test/java/be/seeseemelk/mockbukkit/entity/HumanEntityMockTest.java [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+class HumanEntityMockTest
+{
+
+	private static final int[] expRequired =
+			{
+					7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 42, 47, 52, 57, 62, 67, 72, 77, 82, 87, 92, 97, 102,
+					107, 112, 121, 130, 139, 148, 157, 166, 175, 184, 193
+			};
+	ServerMock server;
+	HumanEntityMock human;
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/entity/HumanEntityMockTest.java]:
```suggestion
			};

	private ServerMock server;
	private HumanEntityMock human;
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java [hunk 12]
```diff
@@ -31,5 +141,396 @@ public void setLastDeathLocation(@Nullable Location location)
...
+		throw new UnimplementedOperationException();
+	}
+
+	@Override
+	public @Nullable InventoryView openStonecutter(@Nullable Location location, boolean force)
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+	@Override
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java]:
```suggestion
	}

	@Override
```

**Review by Insprill — CHANGES_REQUESTED** [review]

**Review by Insprill — APPROVED** [review]

**Pull request merged**